### PR TITLE
doc: add hint why ServiceMonitor might not be in Service Discovery

### PR DESCRIPTION
**Pull request by LittleFox94**
## Description

Apparently ServiceMonitors will not be added to the Service Discovery page if their selector does not match any Service. This is a bit counter-intuitive, so let's just add this to the troubleshooting docs.

I found one other instance of this being confusing on Slack, which helped me figure my problem out: https://kubernetes.slack.com/archives/CFFDS2Z7F/p1664187585784579?thread_ts=1663922438.028919&cid=CFFDS2Z7F

## Type of change

- [ ] `CHANGE` (fix or feature that would cause existing functionality to not work as expected)
- [ ] `FEATURE` (non-breaking change which adds functionality)
- [ ] `BUGFIX` (non-breaking change which fixes an issue)
- [x] `ENHANCEMENT` (non-breaking change which improves existing functionality)
- [ ] `NONE` (if none of the other choices apply. Example, tooling, build system, CI, docs, etc.)

## Changelog entry

```release-note
add hint to troubleshooting docs why a ServiceMonitor might not show up on the Service Discovery page
```

### CODE REVIEW

File: Documentation/troubleshooting.md [hunk 1]
```diff
@@ -63,6 +63,10 @@ Note: The `ServiceMonitor` references a `Service` (not a `Deployment`, or a `Pod
 kubectl -n monitoring get secret prometheus-k8s -ojson | jq -r '.data["prometheus.yaml.gz"]' | base64 -d | gunzip | grep "my-service-monitor"
 ```
 
+#### It is in the configuration but not on the Service Discovery page
+
+ServiceMonitors pointing to a Services that do not exist (e.g. nothing matching `.spec.selector`) will lead to this ServiceMonitor not being added to the Service Discovery page. Check if you can find any Service with the selector you configured.
+
```
Comment (simonpasquier) [inline Documentation/troubleshooting.md:69]: suggestion: we can add the following command which would work for service monitors that don't work cross-namespace and use "simple" matchLabels.

```bash
SERVICE_MONITOR_NAME="<to be replaced>" kubectl get services -l "$(kubectl get servicemonitors $SERVICE_MONITOR_NAME -o jsonpath="{.spec.selector.matchLabels}" | jq -cr 'to_entries | map(.key + "=" + .value) | join(",")')"
````

File: Documentation/troubleshooting.md [hunk 2]
```diff
@@ -63,6 +63,10 @@ Note: The `ServiceMonitor` references a `Service` (not a `Deployment`, or a `Pod
 kubectl -n monitoring get secret prometheus-k8s -ojson | jq -r '.data["prometheus.yaml.gz"]' | base64 -d | gunzip | grep "my-service-monitor"
 ```
 
+#### It is in the configuration but not on the Service Discovery page
+
+ServiceMonitors pointing to a Services that do not exist (e.g. nothing matching `.spec.selector`) will lead to this ServiceMonitor not being added to the Service Discovery page. Check if you can find any Service with the selector you configured.
```
Comment (simonpasquier) [inline Documentation/troubleshooting.md:68]: (nit)
```suggestion
ServiceMonitors pointing to Services that do not exist (e.g. nothing matching `.spec.selector`) will lead to this ServiceMonitor not being added to the Service Discovery page. Check if you can find any Service with the selector you configured.
```

**Review by simonpasquier — COMMENTED** [review]
Thanks! suggested 2 small improvements.

File: Documentation/troubleshooting.md (hunk 1 above)
Comment (LittleFox94) [reply Documentation/troubleshooting.md:75]: added with some describing text, but without depending on any particular shell or `jq` being present

Comment (simonpasquier) [reply Documentation/troubleshooting.md:77]: Nice!

**Review by simonpasquier — APPROVED** [review]
Thanks!

**Pull request merged**